packages/enzyme/src/ReactWrapper.js

```diff
@@ -802,6 +802,27 @@ class ReactWrapper {
     return this.props()[propName];
   }
 
+  /**
+   * Used to invoke a function prop.
+   * Will invoke an function prop and return its value.
+   *
+   * @param {String} propName
+   * @returns {Any}
+   */
+  invoke(propName) {
+    return this.single('invoke', () => {
+      const handler = this.prop(propName);
+      if (typeof handler !== 'function') {
+        throw new TypeError('ReactWrapper::invoke() expects a function prop name as its first argument');
+      }
+      return (...args) => {
+        const response = handler.apply(this, args);
+        this[ROOT].update();
```

Comment:
why is this needed?

Reply:
@ljharb 
The purpose is to reflect the updates into the ShallowWrapper instance if the `handler` calls `setState`.

It's the same as what `simulate()` is doing.

https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ShallowWrapper.js#L999

But I've covered the most case by this PR so this might be unnecessary.

https://github.com/airbnb/enzyme/pull/1499